pkg/mimirtool/commands/template_render.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/prometheus/alertmanager/template"
+
+	MimirAlertManager "github.com/grafana/mimir/pkg/alertmanager"
+)
+
+var defaultData = template.Data{
+	Receiver: "receiver",
+	Status:   "alertstatus",
+	Alerts: template.Alerts{
+		template.Alert{
+			Status: "alertstatus",
+			Labels: template.KV{
+				"label1":          "value1",
+				"label2":          "value2",
+				"instance":        "foo.bar:1234",
+				"commonlabelkey1": "commonlabelvalue1",
+				"commonlabelkey2": "commonlabelvalue2",
+			},
+			Annotations: template.KV{
+				"annotation1":          "value1",
+				"annotation2":          "value2",
+				"commonannotationkey1": "commonannotationvalue1",
+				"commonannotationkey2": "commonannotationvalue2",
+			},
+			StartsAt:     time.Now().Add(-5 * time.Minute),
+			EndsAt:       time.Now(),
+			GeneratorURL: "https://generatorurl.com",
+			Fingerprint:  "fingerprint1",
+		},
+		template.Alert{
+			Status: "alertstatus",
+			Labels: template.KV{
+				"foo":             "bar",
+				"baz":             "qux",
+				"commonlabelkey1": "commonlabelvalue1",
+				"commonlabelkey2": "commonlabelvalue2",
+			},
+			Annotations: template.KV{
+				"aaa":                  "bbb",
+				"ccc":                  "ddd",
+				"commonannotationkey1": "commonannotationvalue1",
+				"commonannotationkey2": "commonannotationvalue2",
+			},
+			StartsAt:     time.Now().Add(-10 * time.Minute),
+			EndsAt:       time.Now(),
+			GeneratorURL: "https://generatorurl.com",
+			Fingerprint:  "fingerprint2",
+		},
+	},
+	GroupLabels: template.KV{
+		"grouplabelkey1": "grouplabelvalue1",
+		"grouplabelkey2": "grouplabelvalue2",
+	},
+	CommonLabels: template.KV{
+		"alertname":       "AlertNameExample",
+		"customer":        "testing_purpose",
+		"environment":     "lab",
+		"commonlabelkey1": "commonlabelvalue1",
+		"commonlabelkey2": "commonlabelvalue2",
+	},
+	CommonAnnotations: template.KV{
+		"commonannotationkey1": "commonannotationvalue1",
+		"commonannotationkey2": "commonannotationvalue2",
+	},
+	ExternalURL: "https://example.com",
+}
+
+// TemplateRenderCmd Render a given definition in a template file to standard output
+type TemplateRenderCmd struct {
+	templateFilesGlobs []string
+	templateType       string
+	templateText       string
+	templateData       *os.File
+	tenantID           string // Needed in the function WithCustomFunctions
+}
+
+func (cmd *TemplateRenderCmd) render(_ *kingpin.ParseContext) error {
+	rendered, err := TemplateRender(cmd)
+	if err != nil {
+		return err
+	}
+	fmt.Print(rendered)
+	return nil
+}
+
+func TemplateRender(cmd *TemplateRenderCmd) (string, error) {
```

Comment:
This code looks very similar to the Prometheus version. I think we need to include license headers, that reference the Prometheus code and license/copyright, to this file. cc @RichiH

Reply:
Yes sure

Reply:
Is it good for you ?

Reply:
Perfect, thanks!

Reply:
Do i need to squash ?

Reply:
Nope, that'll be done when merging.